onadata/libs/utils/logger_tools.py

```diff
@@ -549,17 +540,19 @@ def create_instance(
     xform = get_xform_from_submission(xml, username, uuid, request=request)
     check_submission_permissions(request, xform)
     check_encrypted_submission(xml, xform)
-    checksum = sha256(xml).hexdigest()
     new_uuid = get_uuid_from_xml(xml)
-    filtered_instances = get_filtered_instances(uuid=new_uuid, xform_id=xform.pk)
-    existing_instance = get_first_record(filtered_instances.only("id"))
-    if existing_instance and (new_uuid or existing_instance.xform.has_start_time):
+    duplicate_exists = Instance.objects.filter(
+        xform_id=xform.pk, uuid=new_uuid
+    ).exists()
+
+    if duplicate_exists:
+        duplicate_instance = Instance.objects.get(xform_id=xform.pk, uuid=new_uuid)
```

Comment:
Can we switch to a try ..except .. else  block and avoid the [TOCTOU](https://en.wikipedia.org/wiki/Time-of-check_to_time-of-use) bug?